### ISSUE TITLE
[no-magic-numbers] Static class fields of number type are marked as violation

### ISSUE THREAD

**Issue by TokugawaTakesi** [typescript-eslint/typescript-eslint#945]
## In TSLint, "no-magic-numbers" rule works as expected:

![image](https://user-images.githubusercontent.com/41653501/64313861-fb2df980-cfe7-11e9-9e3c-95365dd9a5ae.png)

The `5` has been marked as magic number, but 3000 is ignored because it semantically associated will static class field name.

**tslint.yaml**

```yaml
rules:
  no-magic-numbers:
    - true
    - allowed-numbers:
      - 0
      - 0.5
      - 1
      - 2
      - 100
```
## "@typescript-eslint/no-magic-numbers" rule of type
[typescript-eslint-no-magic-numbers.zip](https://github.com/typescript-eslint/typescript-eslint/files/3577755/typescript-eslint-no-magic-numbers.zip)
script-eslint malfunction: 

![image](https://user-images.githubusercontent.com/41653501/64316457-da69a200-cfef-11e9-96d3-c5cc4e6b82d2.png)

**.eslintrc.yaml**

```yaml
parser: "@typescript-eslint/parser"
parserOptions:
  sourceType: module
  project: ./tsconfig.json
  tsconfigRootDir: ./

env:
  browser: true
  node: true   
  mocha: true


plugins:
  - "@typescript-eslint"

rules:

  "@typescript-eslint/no-magic-numbers":
    - error
    - ignore:
      - 0
      - 0.5
      - 1
      - 2
      - 100
      enforceConst: true
```

**Repro**

Open the attached project in the IntelliJ IDEA family IDE and enable ESLint inspection to see the warnings as in screenshots above.

```JSON
{
  "rules": {
    "@typescript-eslint/no-magic-numbers": [
       "error",
       {  "ignore": [ 0, 0.5, 1, 2, 100 ], "enforceConst": true }
    ]
  }
}
```

```TS
class Example {

  private static testNumber: number = 3000;

  private static testMethod(): void {
    console.log(5);
  }
}
```

**Expected Result**

In `private static testNumber: number = 3000;`, "3000" must be ignored by typescript-eslint.

**Actual Result**
In `private static testNumber: number = 3000;`, typescript-eslint marks "3000" as "@typescript-eslint/no-magic-numbers" violation.

**Additional Info**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.1.0` |
| `@typescript-eslint/parser`        | `2.1.0` |
| `TypeScript`                       | `3.6.2` |
| `ESLint`                           | `6.3.0` |
| `node`                             | `10.15.3` |
| `npm`                              | `6.11.3` |

**Comment by bradzacher** [comment]
I don't think we'll be adding support for non-readonly properties.


See #934 for readonly prop support.

**Comment by TokugawaTakesi** [comment]
> I don't think we'll be adding support for non-readonly properties.
If so, it means than typescript/eslint will not become to replacement for TSLint.
https://github.com/palantir/tslint/issues/4534
